src/test/java/com/google/sps/AlgorithmTest.java

```diff
@@ -0,0 +1,293 @@
+    placeToVisitB.setProperty("crowd-rating", 4);
+    Entity placeToVisitC = new Entity("Recommendation");
+    placeToVisitC.setProperty("place-name", "Place To Visit C");
+    placeToVisitC.setProperty("latitude", 40.4);
+    placeToVisitC.setProperty("longitude", -0.4);
+    placeToVisitC.setProperty("category", "places-to-visit");
+    placeToVisitC.setProperty("description", "A nice place to visit.");
+    placeToVisitC.setProperty("cost-rating", 4);
+    placeToVisitC.setProperty("crowd-rating", 5);
+    Entity placeToVisitD = new Entity("Recommendation");
+    placeToVisitD.setProperty("place-name", "Place To Visit D");
+    placeToVisitD.setProperty("latitude", 40.5);
+    placeToVisitD.setProperty("longitude", -0.5);
+    placeToVisitD.setProperty("category", "places-to-visit");
+    placeToVisitD.setProperty("description", "An interesting place to visit.");
+    placeToVisitD.setProperty("cost-rating", 2);
+    placeToVisitD.setProperty("crowd-rating", 3);
+    Entity placeToVisitE = new Entity("Recommendation");
+    placeToVisitE.setProperty("place-name", "Place To Visit E");
+    placeToVisitE.setProperty("latitude", 40.6);
+    placeToVisitE.setProperty("longitude", -0.6);
+    placeToVisitE.setProperty("category", "places-to-visit");
+    placeToVisitE.setProperty("description", "A charming place to visit.");
+    placeToVisitE.setProperty("cost-rating", 1);
+    placeToVisitE.setProperty("crowd-rating", 2);
+    Entity placeToVisitF = new Entity("Recommendation");
+    placeToVisitF.setProperty("place-name", "Place To Visit F");
+    placeToVisitF.setProperty("latitude", 40.7);
+    placeToVisitF.setProperty("longitude", -0.7);
+    placeToVisitF.setProperty("category", "places-to-visit");
+    placeToVisitF.setProperty("description", "A wonderful place to visit.");
+    placeToVisitF.setProperty("cost-rating", 2);
+    placeToVisitF.setProperty("crowd-rating", 1);
+    // Store recommendations in datastore
+    ds.put(restaurantA);
+    ds.put(restaurantB);
+    ds.put(restaurantC);
+    ds.put(restaurantD);
+    ds.put(restaurantE);
+    ds.put(studyPlaceA);
+    ds.put(studyPlaceB);
+    ds.put(placeToVisitA);
+    ds.put(placeToVisitB);
+    ds.put(placeToVisitC);
+    ds.put(placeToVisitD);
+    ds.put(placeToVisitE);
+    ds.put(placeToVisitF);
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void testAllFiveReturned() throws Exception{
+    // When there are exactly five recommendations of that category, the algorithm should return  
+    // all 5 recommendations. The coordinate formed by (cost-rating, crowd-rating) should be closest
+    // to the coordinate of user preference for the first recommendation, then respectively for 
+    // the remaining 4 (lowest Euclidean distance)
+
+    // Mock request and mock response
+    HttpServletRequest request = mock(HttpServletRequest.class);
+    HttpServletResponse response = mock(HttpServletResponse.class);
```

Comment:
Do we need to mock these objects? Could we simply create these objects?

Reply:
I'm not sure to be honest - I used mocking as practise but perhaps we could have just made them instead.